MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -109,17 +111,41 @@ private bool IsVaapiSupported(EncodingJobInfo state)
         private bool IsCudaSupported()
         {
             return _mediaEncoder.SupportsHwaccel("cuda")
-                   && _mediaEncoder.SupportsFilter("scale_cuda", null)
-                   && _mediaEncoder.SupportsFilter("yadif_cuda", null);
+                   && _mediaEncoder.SupportsFilter("scale_cuda")
+                   && _mediaEncoder.SupportsFilter("yadif_cuda")
+                   && _mediaEncoder.SupportsFilter("hwupload_cuda");
         }
 
-        private bool IsTonemappingSupported(EncodingJobInfo state, EncodingOptions options)
+        private bool IsOpenclTonemappingSupported(EncodingJobInfo state, EncodingOptions options)
         {
             var videoStream = state.VideoStream;
+            if (videoStream == null)
+            {
+                return false;
+            }
+
             return IsColorDepth10(state)
                    && _mediaEncoder.SupportsHwaccel("opencl")
+                   && _mediaEncoder.SupportsFilter("tonemap_opencl")
                    && options.EnableTonemapping
-                   && string.Equals(videoStream.VideoRange, "HDR", StringComparison.OrdinalIgnoreCase);
+                   && (string.Equals(videoStream.ColorTransfer, "smpte2084", StringComparison.OrdinalIgnoreCase)
```

Comment:
These should be checked first as they are simple string comparisons, which can short circuit the expression